package.json

```diff
@@ -59,14 +59,20 @@
   "bugs": {
     "url": "https://github.com/softonic/axios-retry/issues"
   },
-  "types": "./index.d.ts",
-  "main": "index.js",
+  "types": "lib/esm/index.d.ts",
+  "main": "index.cjs",
   "module": "lib/esm/index.js",
   "exports": {
     ".": {
-      "types": "./index.d.ts",
-      "import": "./lib/esm/index.js",
-      "require": "./index.js"
+      "import": {
+        "types": "./lib/esm/index.d.ts",
+        "default": "./lib/esm/index.js"
+      },
+      "require": {
+        "types": "./lib/cjs/index.d.ts",
+        "default": "./index.cjs"
+      },
```

Comment:
By using [`Conditional exports`](https://nodejs.org/api/packages.html#conditional-exports), the type definition of the imported axios will be switched within the type definition in each case of CommonJS and ES Module. This eliminates compilation errors.

Incidentally, I don't think there will be any impact on the existing use of JavaScript. Because `Conditional exports` was already in use.